### PR TITLE
fix(oracle, clickhouse): ensure `port` is captured in `_from_url` implementation

### PR DESCRIPTION
**Pull request by grieve54706**
## Description of changes
Fix oracle and clickhouse missing `port` in `_from_url`.

### CODE REVIEW

File: ibis/backends/clickhouse/tests/test_client.py [hunk 1]
```diff
@@ -359,3 +365,10 @@ def test_password_with_bracket():
         cc.driver.exceptions.DatabaseError, match="password is incorrect"
     ):
         ibis.clickhouse.connect(host=host, user=user, port=port, password=quoted_pass)
+
+
+def test_invalid_port(con):
+    port = 9999
```
Comment (gforsyth) [inline ibis/backends/clickhouse/tests/test_client.py:371]: Can we add a successful connection using the correct port at the beginning of this test to show that the port is being correctly parsed?

**Review by gforsyth — COMMENTED** [review]
Thanks for putting this in @grieve54706 !

One small request around checking that the port parsing is working as expected (and to contrast it to the behavior when the port is incorrect)

File: ibis/backends/oracle/tests/test_client.py [hunk 2]
```diff
@@ -77,3 +83,13 @@ def test_from_url(con):
...
+
+
+def test_invalid_port(con):
+    port = 9999
+    url = f"oracle://{ORACLE_USER}:{ORACLE_PASS}@{ORACLE_HOST}:{port}/IBIS_TESTING"
+    with pytest.raises(
+        oracledb.OperationalError,
+        match="DPY-6005: cannot connect to database",
+    ):
+        ibis.connect(url)
```
[the remark this answers is not in the document]
Comment (grieve54706) [reply ibis/backends/oracle/tests/test_client.py:95]: I think the `test_from_url` is enough proof. WDYT?

File: ibis/backends/clickhouse/tests/test_client.py (hunk 1 above)
Comment (grieve54706) [reply ibis/backends/clickhouse/tests/test_client.py:378]: Nice idea. I will add a successful test case. Thanks.

**Review by gforsyth — APPROVED** [review]
Awesome!  Thanks for putting this together @grieve54706 !

**Comment by grieve54706** [comment]
Thanks @gforsyth. Thank you for your time.

**Pull request merged**